docs/training/apis-howto/api-core-mnist-ug.rst

```diff
@@ -0,0 +1,573 @@
+-  Modify a script and an experiment configuration file
+-  Understand how to convert model code
+-  Use the Core API to train a model
+
+***************
+ Prerequisites
+***************
+
+**Required**
+
+-  A Determined cluster
+
+**Recommended**
+
+-  :doc:`Quickstart for Model Developers </quickstart-mdldev>`
+
+*****************************************************
+ Step 1: Get the Tutorial Files & Run the Experiment
+*****************************************************
+
+To run an experiment, you need, at minimum, a script and an experiment configuration (YAML) file.
+
+Create a new directory.
+
+Get the files from the :download:`core_api_pytorch_mnist.tgz </examples/core_api_pytorch_mnist.tgz>`
+download or from the `Github repository
+<https://github.com/determined-ai/determined/tree/master/examples/tutorials/core_api_pytorch_mnist>`_.
+
+.. note::
+
+   Throughout this user guide, we’ll show you how to make modifications to your script. After each
+   step, we’ll re-run our experiment using the appropriate script and its accompanying experiment
+   configuration (YAML) file.
+
+In this initial step, we’ll run our experiment using the ``model_def.py`` script and its
+accompanying ``const.yaml`` experiment configuration file.
+
+CD into the directory and run this command:
+
+.. code::
+
+   det e create const.yaml . -f
+
+Open the Determined WebUI by navigating to ``http://localhost:8080/``. Accept the default determined
+username, leave the password empty, and click **Sign In**.
+
+In the WebUI, select your experiment, and then navigate to the **Logs** tab.
+
+************************
+ Step 2: Report Metrics
+************************
+
+To report training and validation metrics to the Determined master, we’ll add a few lines of code to
+our script. More specifically, we'll create a ``core.Context`` object to allow interaction with the
+master. Then, we'll pass the ``core_context`` as an argument into ``main()``, ``train()``, and
+``test()`` and modify the function headers accordingly.
+
+To run our experiment, we'll use the ``model_def_metrics.py script`` and its accompanying
+``metrics.yaml`` experiment configuration file.
+
+Begin by importing Determined:
+
+.. code::
+
+   import determined as det
+
+.. image:: /assets/images/model-def-metrics-script.png
+
+Step 2.1: Modify the Main Loop
+==============================
+
+#. Modification of __main__ loop to include core_context
+
+Modify the __main__loop to include ``core_context``:
+
+.. note::
+
+   Refer to line 211 in ``model_def_metrics.py``
+
+.. code::
+
+   if __name__ == "__main__":
+
+       # NEW: Establish new determined.core.Context and pass to main function
+       with det.core.init() as core_context:
+           main(core_context=core_context)
+
+Step 2.2: Modify the Train Method
+=================================
+
+Modify the train() method by passing ``core_context`` and adding
+``train.report_training_metrics()``:
```

Comment:
From a software perspective, it's confusing to say "adding `train.report_training_metrics()`" because you mention the name `train()` in the same sentence, so mentally if I'm parsing this it sounds like the I'm trying to access the `.report_training_metrics` attribute of the `train()` function.

Basically, you should either refer to this call formally starting with the core context (`core_context.train.report_training_metrics()`) or informally using just the final name (`report_training_metrics()`), but not anything in between I think.

Reply:
correction:

Modify the train() method by adding core_context.train.report_training_metrics():